hazelcast-jet-s3/src/main/java/com/hazelcast/jet/s3/S3Sinks.java

```diff
@@ -161,7 +167,7 @@ private void close() {
 
 
         private String nextKey() {
-            return processorIndex + "-" + (++objectCounter);
+            return prefix + processorIndex + "-" + (++objectCounter);
```

Comment:
how about adding `/` between prefix and processorIndex
this way s3 will create a folder inside the bucket group all objects under that folder

Reply:
Then may be we need to call it folder rather than a prefix if we take that approach ?

Reply:
yes, a `folder` or `directory` maybe

Reply:
@cangencer what is your take on this ?

Reply:
I guess S3 web UI using `folder`

Reply:
hmm I think s3 terminology is prefix, so I'd prefer that, and it's used in source as well. Also you can put like `folder/file-` and then the files be called `folder/file-0-1` etc